### PR TITLE
M3-3670 Making modals and drawers more accessible

### PR DESCRIPTION
**Pull request by alioso**
## M3-3670 Making modals and drawers more accessible

Added necessary markup and new component (modal title, in order to implement focus) to improve accessibility of modal windows 

## Type of Change
- Non breaking change ('update', 'change')

### CODE REVIEW

File: packages/manager/src/components/DialogTitle/DialogTitle.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      tabIndex={-1}
+      className={className}
+      ref={dialogTitile}
+    >
+      {title}
+    </DialogTitle>
+  );
+};
+
+export default compose<any, any>(RenderGuard)(_DialogTitle);
```
Comment (Jskobos) [inline packages/manager/src/components/DialogTitle/DialogTitle.tsx]: Probably don't need this `compose`, but if it stays we should type it better

File: packages/manager/src/components/DialogTitle/DialogTitle.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+// Accessibility Feature:
+// Focus on modal title on component mount
+
+const _DialogTitle: React.FC<Props> = props => {
+  const dialogTitile = React.useRef<HTMLDivElement>(null);
+  const { className, title } = props;
+
+  React.useEffect(() => {
+    if (dialogTitile.current !== null) {
```
Comment (Jskobos) [inline packages/manager/src/components/DialogTitle/DialogTitle.tsx]: Titile

File: packages/manager/src/components/DialogTitle/DialogTitle.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+interface Props {
+  title: string;
+  className?: string;
+}
+
+// Accessibility Feature:
+// Focus on modal title on component mount
+
+const _DialogTitle: React.FC<Props> = props => {
+  const dialogTitile = React.useRef<HTMLDivElement>(null);
```
Comment (Jskobos) [inline packages/manager/src/components/DialogTitle/DialogTitle.tsx]: Titile

**Comment by alioso** [comment]
@Jskobos addressed, also removed renderGuard and compose from the H1Header component to be consistent

**Pull request merged**